engine/src/api/services/llm_client.py

```
@@ -93,6 +233,134 @@ async def chat_completion(
+
+                # Handle different response types
+                if isinstance(content, dict):
+                    # Some models might return a parsed dict directly
+                    result = content
+                elif isinstance(content, str):
+                    try:
+                        # Parse JSON response
+                        result = json.loads(content)
+                    except json.JSONDecodeError as e:
+                        logger.error(f"Error parsing JSON response: {str(e)}")
+                        raise ValueError(f"LLM returned invalid JSON: {content}")
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)